arangod/Aql/TraversalNode.cpp

```diff
@@ -1176,11 +1176,6 @@ void TraversalNode::prepareOptions() {
 
   TraverserOptions* opts = this->TraversalNode::options();
   TRI_ASSERT(opts != nullptr);
-  /*
-   * HACK: DO NOT use other indexes for smart BFS. Otherwise, this will produce
-   * wrong results.
-   */
-  bool onlyEdgeIndexes = this->isSmart() && opts->isUseBreadthFirst();
```

Comment:
This is an old fix that became unnecessary due to the graph refactoring a few years ago. It essentially disabled the usage of non-edge indexes for smart graph bfs.